codec-dns/src/main/java/io/netty/handler/codec/dns/DatagramDnsResponseDecoder.java

```diff
@@ -55,7 +55,10 @@ protected DnsResponse newResponse(InetSocketAddress sender, InetSocketAddress re
 
     @Override
     protected void decode(ChannelHandlerContext ctx, DatagramPacket packet, List<Object> out) throws Exception {
-        out.add(decodeResponse(ctx, packet));
+        // If the content is not readable we should just drop it on the floor as there is nothing we can do with it.
+        if (packet.content().isReadable()) {
```

Comment:
Dropped packets need to be released

Reply:
no... `MessageToMessageDecoder` will release by itself.